text.cabal

```diff
@@ -76,13 +76,26 @@ flag developer
 flag simdutf
   description: use simdutf library, causes Data.Text.Internal.Validate.Simd to be exposed
   default: True
-  manual: True
+
+flag pure-haskell
+  description: Don't use text's standard C routines
+    When this flag is true, text will use pure Haskell variants of the
+    routines. This is not recommended except for use with GHC's JavaScript
+    backend.
+  default: False
```

Comment:
This flag must be manual.

If `arch(javascript)`, make it no-op. But for non-`arch(javascript)` setups the choice matters, and it's very bad to leave it to the solver to decide.